tests/integration/concrete/withdrawal-manager/remove-shares/removeShares.t.sol

```diff
@@ -1,22 +1,53 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity ^0.8.19;
 
+import { Errors } from "contracts/libraries/Errors.sol";
+
 import { WithdrawalManager_Integration_Shared_Test } from "../../../shared/withdrawal-manager/WithdrawalManager.t.sol";
 
 contract removeShares_Integration_Concrete_Test is WithdrawalManager_Integration_Shared_Test {
+    modifier notOverRemove() {
+        _;
+    }
+
     function setUp() public virtual override(WithdrawalManager_Integration_Shared_Test) {
         WithdrawalManager_Integration_Shared_Test.setUp();
+
+        addDefaultShares();
+    }
+
+    function test_RevertWhen_NotPoolConfigurator() external {
+        changePrank(users.caller);
+        vm.expectRevert(abi.encodeWithSelector(Errors.NotPoolConfigurator.selector, users.caller));
+
+        removeDefaultShares();
     }
 
-    function test_removeShares() public whenCallerPoolConfigurator {
+    function test_RevertWhen_WithdrawalPending() external whenCallerPoolConfigurator {
+        vm.expectRevert(abi.encodeWithSelector(Errors.WithdrawalManager_WithdrawalPending.selector, users.receiver));
+        removeDefaultShares();
+    }
+
+    function test_RevertWhen_Overremove() external whenCallerPoolConfigurator notWithdrawalPending {
```

Comment:
Please make the error name more intuitive, e.g. NotEnoughShares, and change the corresponding test name